satpy/readers/mviri_l1b_fiduceo_nc.py

```diff
@@ -0,0 +1,547 @@
+
+        pdict = {
+            'ssp_lon': self.projection_longitude,
+            'a': EQUATOR_RADIUS,
+            'b': POLE_RADIUS,
+            'h': ALTITUDE,
+            'units': 'm',
+            'loff': loff - im_size,
+            'coff': coff,
+            'lfac': -lfac,
+            'cfac': -cfac,
+            'nlines': im_size,
+            'ncols': im_size,
+            'scandir': 'S2N',  # Reference: [PUG] section 2.
+            'p_id': 'geos_mviri',
+            'a_name': 'geos_mviri',
+            'a_desc': 'MVIRI Geostationary Projection'
+        }
+        extent = get_area_extent(pdict)
+        area_def = get_area_definition(pdict, extent)
+        return area_def
+
+    def calibrate(self, ds, channel, calibration):
+        """Calibrate the given dataset."""
+        if channel == 'VIS':
+            return self._calibrate_vis(ds, calibration)
+        elif channel in ['WV', 'IR']:
+            return self._calibrate_ir_wv(ds, channel, calibration)
+        else:
+            raise KeyError('Don\'t know how to calibrate channel {}'.format(
+                channel))
+
+    @abc.abstractmethod
+    def _calibrate_vis(self, ds, calibration):
+        """Calibrate VIS channel. To be implemented by subclasses."""
+        raise NotImplementedError
+
+    def _update_refl_attrs(self, refl):
+        """Update attributes of reflectance datasets."""
+        refl.attrs['sun_earth_distance_correction_applied'] = True
+        refl.attrs['sun_earth_distance_correction_factor'] = self.nc[
+            'distance_sun_earth'].values
+        return refl
+
+    def _calibrate_ir_wv(self, ds, channel, calibration):
+        """Calibrate IR and WV channel."""
+        if calibration == 'counts':
+            return ds
+        elif calibration in ('radiance', 'brightness_temperature'):
+            rad = self._ir_wv_counts_to_radiance(ds, channel)
+            if calibration == 'radiance':
+                return rad
+            bt = self._ir_wv_radiance_to_brightness_temperature(rad, channel)
+            return bt
+        else:
+            raise KeyError('Invalid calibration: {}'.format(calibration.name))
+
+    def _ir_wv_counts_to_radiance(self, counts, channel):
+        """Convert IR/WV counts to radiance.
+
+        Reference: [PUG], equations (4.1) and (4.2).
+        """
+        if channel == 'WV':
+            a, b = self.nc['a_wv'], self.nc['b_wv']
+        else:
+            a, b = self.nc['a_ir'], self.nc['b_ir']
+        rad = a + b * counts
+        return rad.where(rad > 0)
+
+    def _ir_wv_radiance_to_brightness_temperature(self, rad, channel):
+        """Convert IR/WV radiance to brightness temperature.
+
+        Reference: [PUG], equations (5.1) and (5.2).
+        """
+        if channel == 'WV':
+            a, b = self.nc['bt_a_wv'], self.nc['bt_b_wv']
+        else:
+            a, b = self.nc['bt_a_ir'], self.nc['bt_b_ir']
+        bt = b / (np.log(rad) - a)
+        return bt.where(bt > 0)
```

Comment:
These two functions look very similar! maybe the coeff fetching could be factorised as a start?

Reply:
Done!